### ISSUE TITLE
Being able to load a glb model from a string

### ISSUE THREAD

**Issue by marcusx2** [atteneder/glTFast#341]
See [this example](https://playground.babylonjs.com/#88CB6A). Apparently it's possible to load a glb from a string. I am not sure if the example uses [gltf embedded](https://docs.blender.org/manual/en/2.80/addons/io_scene_gltf2.html#gltf-embedded-gltf), or if it's something else.

The use case from this is being able to load a glb model at runtime from a [url query string](https://en.wikipedia.org/wiki/Query_string).

In addition, being able to get the string representation of a .glb file.

So .glb -> glb string and glb string -> glb

When generating the glb string it should be **not** prettified by default, but a prettify option would be nice, but not necessary. By prettify I mean showing everything nicely formatted with new lines, etc. Not prettified is everything on a single line with no unnecessary spaces.

Looking forward to hearing back from you @atteneder . I have the feeling this is  already possible, and if not it's probably something that can easily be implemented 🙏

**Comment by atteneder** [comment]
Hi @marcusx2 ,

you can already [load a glTF from a byte array](https://github.com/atteneder/glTFast/blob/main/Documentation%7E/ImportRuntime.md#load-from-byte-array). Converting a string to a byte array is doable as well, so there you go.

Let me know if this helped you and what use-case you realized.

**Comment by atteneder** [comment]
> Oh I see, thanks. What if I load a glb file from the user's computer, and want to convert it to a byte array (or string), is it possible?

Please click the link I already provided in my last answer. It literally does just that.

**Comment by marcusx2** [comment]
Oh I see sorry for the brainfart...when you load a file you already store it in a byte array format, so I just need to convert it to a string.

**Comment by atteneder** [comment]
> Sorry for the brainfart...when you load a file you already store it in a byte array format, so I just need to convert it to a string.

Why? There would be no benefit in converting to a string. It's just that if your data source was a string, you have to convert to a byte array.

One thing I realise now is that `LoadGltfBinary` only loads glTF-Binary files (no JSON based .gltf).

**Comment by marcusx2** [comment]
`Why? There would be no benefit in converting to a string. It's just that if your data source was a string, you have to convert to a byte array.`

If I can send the byte array as a query string in a URL, then I don't need to convert it indeed. I just assumed that a query string only accepts, well, strings, and not byte arrays.

**Comment by atteneder** [comment]
> If I can send the byte array as a query string in a URL, then I don't need to convert it indeed. I just assumed that a query string only accepts, well, strings, and not byte arrays.

Right, got it! thx

**Comment by marcusx2** [comment]
@atteneder hey man, I have just one question about this line `await gltf.LoadGltfBinary(data, new Uri(m_Path));`, what do I do about the path? There is no path technically. I uploaded the glb from the user's computer at runtime, the glb wasn't in the project folder. Can the path be null?

**Comment by atteneder** [comment]
> what do I do about the path? There is no path technically. I uploaded the glb from the user's computer at runtime, the glb wasn't in the project folder. Can the path be null?

TL;DR: Yes, the path might be omitted.

Explanation: Any glTF (even a glb file) could theoretically reference external URIs, which are usually relative paths. In order to resolve those relative paths, glTFast needs the original path. Most glb files are self-contained, so you won't have problems there. Only files with relative external references will fail.

hth

**Comment by marcusx2** [comment]
Hey man, I managed to upload the 3d model and instantiate it on the scene just fine. But the model is pink. I read the documentation, but couldn't figure this out. Could you add some pictures? I want to preload all the shaders. this is what I tried:

![image](https://user-images.githubusercontent.com/5656275/156038288-e3defe2d-25de-4505-8779-599db24b0fee.png)

**Comment by atteneder** [comment]
@marcusx2 I received an email with a comment from you having trouble getting the shaders into the build. The comment is gone, so I guess you solved it and deleted the comment. If you have insights what your problem was and where the documentation is unclear, maybe we could improve them for everyone.
Thanks for considering.

**Comment by marcusx2** [comment]
Yes, I think the documentation is unclear...I solved my issue. In the documentation you talk about the preloaded shaders section. Which is here

![image](https://user-images.githubusercontent.com/5656275/156238757-6b765f47-eda0-49ba-bcc3-43530bd53b02.png)

but this doesn't solve the problem of the pink textures. What solved the problem of the pink textures was adding the shaders to the always included shaders

![image](https://user-images.githubusercontent.com/5656275/156238922-fdab7fcd-e322-496d-8755-6d2fb9f5c706.png)

I think using pictures would be better, a picture is worth a thousand words.
